src/config.rs

```diff
@@ -112,21 +109,29 @@ fn default_pdb_config_section() -> PdbConfigSection {
     }
 }
 
-#[derive(RustcDecodable,RustcEncodable,Debug)]
-pub struct CLIConfig {
-    puppetdb: Option<PdbConfigSection>,
-}
-
-impl CLIConfig {
-    fn load(path: String) -> CLIConfig {
+impl PdbConfigSection {
+    fn load(path: String) -> PdbConfigSection {
         let mut f = File::open(&path).unwrap_or_else(|e| {
             pretty_panic!("Error opening config {:?}: {}", path, e)
         });
         let mut s = String::new();
         if let Err(e) = f.read_to_string(&mut s) {
             pretty_panic!("Error reading from config {:?}: {}", path, e)
         }
-        json::decode(&s).unwrap_or_else(|e| pretty_panic!("Error parsing config {:?}: {}", path, e))
+        let json = json::Json::from_str(&s).unwrap_or_else(|e| pretty_panic!("Error parsing config {:?}: {}", path, e));
+        PdbConfigSection {
+            server_urls:
+            match json.find_path(&["puppetdb", "server_urls"])
+                .unwrap_or(&json::Json::Null).clone() {
+                    json::Json::Array(urls) => Some(urls.into_iter().map(|url| url.as_string().unwrap().to_string()).collect::<Vec<String>>()),
+                    json::Json::String(urls) => Some(split_server_urls(urls)),
+                    json::Json::Null => Some(default_server_urls()),
+                    _ => pretty_panic!("Error parsing config {:?}: server_urls must be an Array or a String", path),
+                },
+            cacert: json.find_path(&["puppetdb", "cacert"]).and_then(|s| s.as_string().and_then(|s| Some(s.to_string()))),
```

Comment:
What's interesting here is the reason you need to add
```
.and_then(|s| Some(s.to_string()))
```
is because `s.as_string()` returns `Option(&str)`. Similarly, line 126 needs to do the same thing. Maybe after this PR we get in a change to switch the structs from String to &str. I have that locally from way back when, probably just worth starting over. But we pay for the conversion here and we're not using the mutability.